### PR TITLE
VEN-719 | Implement table in winter storage area view

### PR DESCRIPTION
**Pull request by ahie**
## Description :sparkles:

Adds missing table contents and functionality to the winter storage area view.

### Closes :no_good_woman:

[VEN-719](https://helsinkisolutionoffice.atlassian.net/browse/VEN-719)

### Automated tests :gear:️

- Updated relevant snapshots

### Manual testing :construction_worker_man:

- Open a winter storage area's page that has places
- The places should be included in the table

## Additional notes :spiral_notepad:

- Also added width and length columns to the table similar to the berths table.
- No "Huolto" column
- changes in `src/customerView` are temporary and properly addressed in #249

### CODE REVIEW

File: src/common/selectHeader/SelectHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import styles from './selectHeader.module.scss';
+
+interface SelectHeaderProps<T> {
+  className?: string;
+  items: T[];
+  selectedItem?: T | null;
+  allLabel: string;
+  editLabel: string;
+  formatter(toFormat: T): string;
```
Comment (santtul) [inline src/common/selectHeader/SelectHeader.tsx:12]:
```suggestion
  formatter(item: T): string;
```

File: src/common/selectHeader/SelectHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  equals(a: T, b: T): boolean;
+  renderer?(item: T): React.ReactNode;
+  onSelect(pier: T | null): void;
```
Comment (santtul) [inline src/common/selectHeader/SelectHeader.tsx:15]:
```suggestion
  onSelect(item: T | null): void;
```

File: src/common/selectHeader/SelectHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  onEdit?(pier: T): void;
```
Comment (santtul) [inline src/common/selectHeader/SelectHeader.tsx:16]:
```suggestion
  onEdit?(item: T): void;
```

File: src/features/winterStorageAreaView/WinterStorageAreaViewTable.tsx [hunk 4]
```diff
@@ -1,29 +1,109 @@
...
+
+        return (
+          <SelectHeader
+            items={sections}
+            selectedItem={selectedSection}
+            allLabel={t('common.table.all')}
+            editLabel={t('common.edit')}
+            formatter={(section) => `${t('winterStorageAreaView.tableHeaders.identifier')} ${section.identifier}`}
+            equals={(a, b) => a.identifier === b.identifier}
+            onSelect={(pier) => props.setFilter('identifier', pier?.identifier)}
```
Comment (santtul) [inline src/features/winterStorageAreaView/WinterStorageAreaViewTable.tsx:93]:
```suggestion
            onSelect={(section) => props.setFilter('identifier', section?.identifier)}
```

File: src/features/winterStorageAreaView/utils.ts [hunk 5]
```diff
@@ -55,6 +55,68 @@ export const getIndividualWinterStorageArea = (
...
+              number: placeEdge.node.number,
+              width: placeEdge.node.width,
+              length: placeEdge.node.length,
+              isActive: placeEdge.node.isActive,
+              leases,
+            },
+          ];
+        }, []) ?? []
+    )
+    .reduce<WinterStoragePlace[]>((acc, arr) => [...acc, ...arr], []);
```
Comment (Nizar-Rahme) [inline src/features/winterStorageAreaView/utils.ts:109]: Chaining `reduce`, `map` and `reduce` seems a bit inefficient, couldn't the preceding logic be packaged in the first `recude`'s callback?

File: src/features/customerView/utils.ts [hunk 6]
```diff
@@ -170,9 +170,9 @@ export const getBills = (profile: CUSTOMER_PROFILE): (BerthBill | WinterStorageB
           {
             status: orderNode.status,
             berthInformation: {
-              number: lease.berth.number,
-              pierIdentifier: lease.berth.pier.properties?.identifier ?? '',
-              harborName: lease.berth.pier.properties?.harbor?.properties?.name ?? '',
+              number: 0,
+              pierIdentifier: '',
+              harborName: '',
```
Comment (Nizar-Rahme) [inline src/features/customerView/utils.ts:175]: Could you add a `TODO` comment here to replace the hardcoded values once they are implemented on the backend?

File: src/features/winterStorageAreaView/utils.ts [hunk 7]
```diff
@@ -55,6 +55,68 @@ export const getIndividualWinterStorageArea = (
...
+      }
+      return [...acc, sectionEdge.node.properties];
+    }, [])
+    .map<WinterStoragePlace[]>(
+      (sectionProperties) =>
+        sectionProperties.places.edges.reduce<WinterStoragePlace[]>((acc, placeEdge) => {
+          if (!placeEdge?.node) {
+            return acc;
+          }
+          const leases = placeEdge?.node?.leases?.edges.reduce<Lease[]>((acc, leaseEdge) => {
```
Comment (santtul) [inline src/features/winterStorageAreaView/utils.ts:75]: The function is quite big, would it make sense to extract this piece as separate function (`getLeases`) ?

File: src/common/selectHeader/SelectHeader.tsx (hunk 1 above)
Comment (ahie) [reply src/common/selectHeader/SelectHeader.tsx]: Done.

File: src/common/selectHeader/SelectHeader.tsx (hunk 2 above)
Comment (ahie) [reply src/common/selectHeader/SelectHeader.tsx]: Done.

File: src/common/selectHeader/SelectHeader.tsx (hunk 3 above)
Comment (ahie) [reply src/common/selectHeader/SelectHeader.tsx]: Done.

File: src/features/winterStorageAreaView/WinterStorageAreaViewTable.tsx (hunk 4 above)
Comment (ahie) [reply src/features/winterStorageAreaView/WinterStorageAreaViewTable.tsx]: Done.

File: src/features/winterStorageAreaView/utils.ts (hunk 5 above)
Comment (ahie) [reply src/features/winterStorageAreaView/utils.ts:117]: With the refactoring done in the latest changes, is this comment still relevant? Mixing the list concatenation logic to the previous callbacks would IMO make the code harder to read.

File: src/features/customerView/utils.ts (hunk 6 above)
Comment (ahie) [reply src/features/customerView/utils.ts:176]: Done.

File: src/features/winterStorageAreaView/utils.ts (hunk 7 above)
Comment (ahie) [reply src/features/winterStorageAreaView/utils.ts]: Done.

File: src/features/winterStorageAreaView/utils.ts (hunk 5 above)
Comment (Nizar-Rahme) [reply src/features/winterStorageAreaView/utils.ts:117]: Looks better!

**Pull request merged**